introduction.md

```diff
@@ -1,98 +1,51 @@
-techniques that allow us to compare and contrast solutions based solely
-on their own characteristics, not the characteristics of the program or
-computer used to implement them.
-
-In the worst case scenario, we may have a problem that is intractable,
-meaning that there is no algorithm that can solve the problem in a
-realistic amount of time. It is important to be able to distinguish
-between those problems that have solutions, those that do not, and those
-where solutions exist but require too much time or other resources to
-work reasonably.
-
-There will often be trade-offs that we will need to identify and decide
-upon. As computer scientists, in addition to our ability to solve
-problems, we will also need to know and understand solution evaluation
-techniques. In the end, there are often many ways to solve a problem.
-Finding a solution and then deciding whether it is a good one are tasks
-that we will do over and over again.
-
+This book is a practical—and we hope, entertaining—introduction to some
+of the most important data structures and algorithms in computer
+science.
+
+We have written the examples in this book in idiomatic Python to help
+convey the *usefulness* of these data structures and algorithms, both
+directly and as models for analogous work that you will do.
+
+Many software engineers are skeptical that this subject matter is
+useful. Our best explanation for such skepticism is that those engineers
+never encountered these topics in a way that was encouraged deep
+understanding. Perhaps they were [unfortunate
+enough](http://v.cx/2010/04/feynman-brazil-education) to participate in
+a computer science program that encouraged rote memorization, or
+subjected themselves to such an experience for the sake of succeeding at
+technical interviews. Or perhaps they became adequate programmers
+without having encountered the content at all, and presume that they
+have reached the limits of their capabilities.
+
+Whatever the case, we’re confident that the usefulness of studying
+algorithms and data structures increases proportionally to the depth at
+which you understand it—a depth that comes through a combination of
+study and application. Those who understand trees in depth are more
+likely to identify tree shapes in their own data, and have the
+confidence to apply a traversal algorithm appropriately. Those who count
+Big-O analysis as a tool in their toolkits not only have a better
+vocabulary for comparing alternative strategies, they are also more
+cognizant overall of the trade offs we implicitly make every day. Those
+who see recursion everywhere see recursion everywhere :)
+
+One trade off that *we* have made is to sacrifice breadth for the sake
+of depth. We cover fewer data structures and algorithms than a typical
+introductory text may cover, in order to focus more on what we consider
+to be important. For instance we do not cover sorting algorithms at all,
```

Comment:
perhaps reassure the reader here by letting them know that they'll probably never have to implement a sorting algorithm